backend/src/main/java/io/github/k_gregory/registry/infrastructure/ResourceNotFoundException.java

```diff
@@ -0,0 +1,9 @@
+package io.github.k_gregory.registry.infrastructure;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
+public class ResourceNotFoundException extends RuntimeException {
```

Comment:
Maybe we shall add here static generic method that takes one parameter, checks if it is null and throws ResourceNotFoundException in that case? It will save some boilerplate code.